### ISSUE TITLE
crash on 3.0.0 sIngle node cluster doesn't come up after hard reboot

### ISSUE THREAD

**Issue by laralar** [lxc/lxd#4628]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 18.04 LTS
 * The output of "lxc info" or if that fails:
   * Kernel version: 
   * LXC version: 3.0.0
   * LXD version: 3.0.0
   * Storage backend in use: btrfs subvolume mounted on /var/lib/lxd, btrfs storage pool as a subvolume of the latter

# Issue description

After reboot LXD daemon doesnt come up, did cluster database got corrupted?

# Steps to reproduce
No idea. what other info should i provide?

# Information to attach

from /var/log/syslog

```
Jun  7 23:24:54 node12 lxd[2610]: lvl=eror msg="Failed to start the daemon: failed to open cluster database: failed to ensure schema: failed to begin transaction: gRPC
BEGIN response error: rpc error: code = Unknown desc = failed to handle BEGIN request: FSM out of sync: timed out enqueuing operation" t=2018-06-07T23:24:54+0530
Jun  7 23:24:54 node12 lxd[2610]: Error: failed to open cluster database: failed to ensure schema: failed to begin transaction: gRPC BEGIN response error: rpc error: co
de = Unknown desc = failed to handle BEGIN request: FSM out of sync: timed out enqueuing operation
Jun  7 23:24:54 node12 lxd[2151]: Error: Get http://unix.socket/1.0: EOF
```

**Comment by stgraber** [comment]
Make sure that the system time is identical on all cluster nodes, this has caused problems to some of our users in the past.

Then on the failing node, try:
 - systemctl stop lxd lxd.socket
 - lxd --debug --group lxd

Note that LXD 3.0.0 in the Ubuntu archive has a number of known issues related to clustering which LXD 3.0.1 is addressing, but we're waiting for the Ubuntu SRU team to review that release...

**Comment by stgraber** [comment]
Hmm, okay so no way for us to figure out what actually happened and if it's one of the issues that has been fixed in 3.0.1.

You can turn a non-clustered node into a cluster and then join nodes to it after the fact, no need to enable it ahead of time. LXD 3.0.1 includes a convenient `lxc cluster enable` subcommand which does that for you.

Over-mounting `/var/lib/lxd` is fine, LXD itself won't mind but it's certainly true that systemd's mount ordering and cleanup code can somewhat easily be confused leading to those delays on shutdown.
I've seen the same kind of issues pop up sometimes even when bindmounts aren't involved, so whether /var/lib/lxd is on a different partition shouldn't matter all that much.

**Comment by laralar** [comment]
Appreciated.

**Comment by freeekanayaka** [comment]
The database is not corrupted. This issue is very likely a dupe of #4485.

**Comment by laralar** [comment]
Running the script automatically made lxd to status activating and /var/log/syslog showing out some data and containers starting.

Thanks

**Comment by laralar** [comment]
Is this going to be part of 3.0.1 or 3.1 ?

I am not using snaps,, default 1804 cloud installation from MAAS deploy so far.  I have a few servers.. so scripting the deb to snap migration will take quite a while and it would probably require manual intervention on around 50 nodes.

**Comment by laralar** [comment]
Not sure if I understood correctly, @stgraber `Note that this will not persist on reboot and that this will no longer be needed after you run a fix LXD` 

I had to re apply the script after reboot.  Thanks.

**Comment by stgraber** [comment]
The fix is in both 3.0.1 and 3.1. Since you're using the deb, your only option will be 3.0.1 which will hopefully show up in Ubuntu updates in the next couple of weeks.
